### DISCUSSION TITLE
compare to phaserjs

### DISCUSSION THREAD

**Discussion by SteveSuv** [excaliburjs/Excalibur#3030] [General]
I am choosing a framework between phaserjs and excaliburjs to develop a 2d web game. I can't figure out which is better, what's the main difference between them.

**Comment by eonarheim** [comment]
Hi @SteveSuv! 

First I'll say you should pick whatever make you happy and productive 😀 

As far as feature differences they are probably roughly the same and you can accomplish similar things in 2D across both. The biggest differentiator is probably the API style, Excalibur is very OOP/TypeScript oriented similar to Unity/Godot. If you like that style you'll be right at home.

Speaking for Excalibur, it is designed with approachability in mind. You can make some [classic favorites](https://excaliburjs.com/docs/getting-started) with just a few lines of code! Excalibur will meet you where you're at, whether you're new to game development or an experienced game maker.

Excalibur has a ton of features to help you make games!

- Lots of documentation at https://excaliburjs.com/!
  - [Samples](https://excaliburjs.com/samples) & [Showcase](https://excaliburjs.com/showcase)
- Objected-Oriented TypeScript first API similar to Unity/Godot if you like these you'll be right at home
- Friendly API based around a theater metaphor of Scenes, Actors, and Actions
  - [Actors](https://excaliburjs.com/docs/actors) come prebuilt everything you need!
- [SpriteSheet](https://excaliburjs.com/docs/spritesheets), [Sprites](https://excaliburjs.com/docs/sprites), and [Animations](https://excaliburjs.com/docs/animation)
- [Text and Font](https://excaliburjs.com/docs/text) support
- Built in Math utilities like [Matrix](https://excaliburjs.com/docs/matrix) and [Vector](https://excaliburjs.com/docs/vector)
- [Collision Detection](https://excaliburjs.com/docs/colliders)
- [Arcade](https://excaliburjs.com/docs/physics#arcade) Style and [Realistic](https://excaliburjs.com/docs/physics#realistic) Physics
- [2D](https://excaliburjs.com/docs/tilemap) and [Isometric](https://excaliburjs.com/docs/isometric) Tilemaps
- Auto-batching draw calls and really tuned draw performance
- Popular Game Dev Tool Support
  * [Tiled maps](https://excaliburjs.com/docs/tiled-plugin)
  * [LDtk levels](https://excaliburjs.com/docs/ldtk-plugin)
  * [Spritefusion levels](https://excaliburjs.com/docs/spritefusion-plugin)
  * [Aseprite image files](https://github.com/excaliburjs/excalibur-aseprite)
  * [JSFXR audio](https://github.com/excaliburjs/sample-jsfxr/)
- Optionally use [Entity-Component-System](https://excaliburjs.com/docs/entity-component-system) under the hood when you need control
- Optionally write [custom WebGL Renderers](https://excaliburjs.com/docs/custom-renderer-plugins)
